src/actions/Entities/NameNoteActions.js

```diff
@@ -43,12 +44,29 @@ const createDefaultNameNote = (nameID = '') => {
   };
 };
 
-const createSurveyNameNote = nameID => (dispatch, getState) => {
-  const name = UIDatabase.get('Name', nameID);
+const getMostRecentPCD = patient => {
+  const [pcdEvent] = UIDatabase.objects('PCDEvents');
+  if (!pcdEvent) return null;
 
+  const { id: pcdEventID } = pcdEvent;
+  const { nameNotes = [] } = patient ?? {};
+  if (!nameNotes.length) return null;
+
+  const filtered = nameNotes.filter(({ patientEventID }) => patientEventID === pcdEventID);
+  if (!filtered.length) return null;
+
+  const sorted = filtered.sort(
+    ({ entryDate: entryDateA }, { entryDate: entryDateB }) =>
+      moment(entryDateB).valueOf() - moment(entryDateA).valueOf()
+  );
+  const [mostRecentPCD] = sorted;
+  return mostRecentPCD;
+};
```

Comment:
This is changed such that it can look up the most recent name note for an existing patient or one from the lookup API.

This will need this mSupply PR https://github.com/sussol/msupply/pull/8280 to work- I haven't tested it with it yet